src/melee.cpp

```diff
@@ -322,11 +292,10 @@ void player::melee_attack(Creature &t, bool allow_special, const matec_id &force
 
         t.on_dodge( this, get_melee() );
 
-        if( !has_active_bionic("bio_cqb") ) {
-            // No practice if you're relying on bio_cqb to fight for you
-            melee_practice( *this, false, unarmed_attack(),
-                            weapon.is_bashing_weapon(), weapon.is_cutting_weapon(),
-                            (weapon.has_flag("SPEAR") || weapon.has_flag("STAB")));
+        // Practice melee and relevant weapon skill (if any) except when using CQB bionic
+        if( !has_active_bionic( "bio_cqb" ) ) {
+            practice( skill_melee, rng( 5, 10 ) );
+            practice( weapon.weap_skill(), rng( 5, 10 ) );
```

Comment:
Splitting into different attacks is good, but making them all "typed" is not. We had this earlier in the development for knives and it lead to them always training only one of the types due to it scaling better at low levels.

Splitting training sounds good.
But it shouldn't be 150% - more two `practice`s which sum up to 150%, but each is at most 100%.
This is because low amount practices are inherently worse when repeated a lot. Each practice drains focus regardless of amount.
Alternatively, if training one type only, it could be trained twice instead of for twice the amount.